src/uct/ib/dc/base/dc_iface.c

```diff
@@ -441,7 +441,8 @@ ucs_status_t uct_dc_iface_fc_handler(uct_rc_iface_t *rc_iface, unsigned qp_num,
         dc_req->super.ep         = &ep->super.super;
         dc_req->dct_num          = imm_data;
         dc_req->lid              = lid;
-        dc_req->sender_ep        = *((uintptr_t*)(hdr + 1));
+        dc_req->sender.ep        = *((uintptr_t*)(hdr + 1));
+        dc_req->sender.gid       = iface->super.super.gid;
```

Comment:
you need to get the GID from the UCT_RC_EP_FC_FLAG_HARD_REQ packet (like ep addr), not local.

Reply:
fixed, thank you